bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/internal/handler/SoulissGatewayHandler.java

```diff
@@ -0,0 +1,324 @@
+            }
+
+            // START SERVER ON DEFAULT PORT - Used for topics
+            if (udpServerDefaultPortRunnableClass == null) {
+                logger.debug("Starting UDP server on Souliss Default Port for Topics (Publish&Subcribe)");
+                datagramSocketDefaultPort = SoulissDatagramSocketFactory.getSocketDatagram(soulissGatewayPort,
+                        this.logger);
+                if (datagramSocketDefaultPort != null) {
+                    udpServerDefaultPortRunnableClass = new SoulissBindingUDPServerJob(datagramSocketDefaultPort,
+                            SoulissBindingNetworkParameters.discoverResult);
+                    // Changes from scheduleAtFixedRate - Luca Calcaterra
+                    scheduler.scheduleWithFixedDelay(udpServerDefaultPortRunnableClass, 100,
+                            SoulissBindingConstants.SERVER_CICLE_IN_MILLIS, TimeUnit.MILLISECONDS);
+                }
+            }
+
+            // START JOB PING
+
+            SoulissGatewayJobPing soulissGatewayJobPingRunnable = new SoulissGatewayJobPing(bridge);
+            // Changes from scheduleAtFixedRate - Luca Calcaterra
+            scheduler.scheduleWithFixedDelay(soulissGatewayJobPingRunnable, 2,
+                    soulissGatewayJobPingRunnable.getPingRefreshInterval(), TimeUnit.SECONDS);
+
+            SoulissGatewayJobSubscription soulissGatewayJobSubscriptionRunnable = new SoulissGatewayJobSubscription(
+                    bridge);
+            // Changes from scheduleAtFixedRate - Luca Calcaterra
+            scheduler.scheduleWithFixedDelay(soulissGatewayJobSubscriptionRunnable, 0,
+                    soulissGatewayJobSubscriptionRunnable.getSubscriptionRefreshInterval(), TimeUnit.MINUTES);
+
+            SoulissGatewayJobHealthy soulissGatewayJobHealthyRunnable = new SoulissGatewayJobHealthy(bridge);
+            // Changes from scheduleAtFixedRate - Luca Calcaterra
+            scheduler.scheduleWithFixedDelay(soulissGatewayJobHealthyRunnable, 5,
+                    soulissGatewayJobHealthyRunnable.gethealthRefreshInterval(), TimeUnit.SECONDS);
+
+            // il ciclo Send è schedulato con la costante
+            // SoulissBindingConstants.SEND_DISPATCHER_MIN_DELAY_cicleInMillis
+            // internamente il ciclo viene rallentato al timer impostato da configurazione (PaperUI o File)
+            SoulissBindingSendDispatcherJob soulissSendDispatcherRunnable = new SoulissBindingSendDispatcherJob(bridge);
+            scheduler.scheduleWithFixedDelay(soulissSendDispatcherRunnable, 15,
+                    SoulissBindingConstants.SEND_DISPATCHER_MIN_DELAY_CYCLE_IN_MILLIS, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private int gwIpByte() {
+        return Integer.parseInt(ipAddressOnLAN.split("\\.")[3]);
+    }
+
+    @Override
+    public void handleRemoval() {
+        SoulissBindingNetworkParameters.removeGateway((byte) gwIpByte());
+        logger.debug("Gateway handler removing");
```

Comment:
I think you want the gateway registered only if the Thing is initialized (i.e. the config is loaded correctly and so on). You could register/deregister these in `initialize()` and `dispose()`.